### ISSUE TITLE
 Length of string field characters

### ISSUE THREAD

**Issue by tarashchuk** [vespa-engine/vespa#31733]
Can you please add function for ranking feature with get Length of string field characters. Also it will be great to have length of all characters of terms or something like this.

**Comment by jobergum** [comment]
Could you add examples to clarify what you are asking for? The length of a string can be computed by you and added as an attribute field to be used in ranking.

```
schema foo {
   document foo {
       field bar type string{}
       field bar_length type int{}
    }
}```

**Comment by tarashchuk** [comment]
@jobergum agree, but we discussed about ability to calculate it in ranking feature without adding a new field.

**Comment by jobergum** [comment]
IMHO; We don't want to add a general feature that keeps track of character length of strings to be accessed in ranking because:

- Things you access randomly in ranking expressions should be in memory (defined with attribute) for meaningful performance  
- Adding character length for all string fields can easily be added by the application on a per field basis.

**Comment by kkraune** [comment]
another idea: add to indexing language, populate a synthetic field with length of input field (in tokens? characters? bytes?)
